File: src/mvg_departures/domain/models/departure.py
"""Departure domain model."""

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Departure:
    """Represents a single departure from a station."""

    time: datetime
    planned_time: datetime
    delay_seconds: int | None
    platform: int | None
    is_realtime: bool
    line: str
    destination: str
    transport_type: str
    icon: str
    is_cancelled: bool
    messages: list[str]
    stop_point_global_id: str | None = (
        None  # Physical stop point identifier (e.g., "de:09162:1108:4:4")
    )

File: spa/src/domain/models/departure.ts
export interface Departure {
  time: Date;
  plannedTime: Date;
  delaySeconds: number | null; // Match Python: delay_seconds: int | None
  platform: string | null;
  isRealtime: boolean;
  line: string;
  destination: string;
  transportType: string;
  icon: string;
  isCancelled: boolean;
  messages: string[];
  stopPointGlobalId: string | null;
}

export function createDeparture(data: {
  time: Date | number;
  plannedTime: Date | number;
  delaySeconds?: number | null;
  platform?: string | null;
  isRealtime?: boolean;
  line: string;
  destination: string;
  transportType: string;
  icon?: string;
  isCancelled?: boolean;
  messages?: string[];
  stopPointGlobalId?: string | null;
}): Departure {
  return {
    time: data.time instanceof Date ? data.time : new Date(data.time),
    plannedTime:
      data.plannedTime instanceof Date
        ? data.plannedTime
        : new Date(data.plannedTime),
    delaySeconds: data.delaySeconds ?? null, // Match Python: defaults to None, not 0
    platform: data.platform ?? null,
    isRealtime: data.isRealtime ?? false,
    line: data.line,
    destination: data.destination,
    transportType: data.transportType,
    icon: data.icon ?? "",
    isCancelled: data.isCancelled ?? false,
    messages: data.messages ?? [],
    stopPointGlobalId: data.stopPointGlobalId ?? null,
  };
}
